common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -899,6 +905,90 @@ public static ClassLoader getSystemClassLoader() {
         }
     }
 
+    /**
+     * Return the size of a cache line for the running platform or {@code -1} if is unknown
+     * or could not be detected.
+     */
+    public static int cacheLineSize() {
+        return CACHE_LINE_SIZE;
+    }
+
+    private static int cacheLineSize0() {
+        return AccessController.doPrivileged(new PrivilegedAction<Integer>() {
+            @Override
+            public Integer run() {
+                if (IS_LINUX) {
+                    BufferedReader in = null;
+                    try {
+                        in = new BufferedReader(new InputStreamReader(new FileInputStream("/proc/cpuinfo")));
+                        String line;
+                        while ((line = in.readLine()) != null) {
+                            if (line.startsWith("cache_alignment")) {
+                                String[] parts = line.split(":");
+                                return Integer.parseInt(parts[1].trim());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method io.netty.util.internal.PlatformDependent$1.run() converts String to boxed primitive using excessive boxing [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOXING_VALUEOF)